### PR TITLE
feat: Use ensureKonnectorFolder from cozy-client

### PR DESCRIPTION
**Pull request by doubleface**
And also update the destination folder id saved in the trigger when
needed.

Upgrade cozy-client to 41.4.1 to be able to update the trigger and to
have ensureKonnectorFolder function.

#### Checklist

Before merging this PR, the following things must have been done:

* [ ] Faithful integration of the mockups at all screen sizes
* [ ] Tested on iOS
* [X] Tested on Android
* [X] Localized in English and French
* [ ] All changes have test coverage
* [ ] Updated README & CHANGELOG, if necessary

### CODE REVIEW

File: src/libs/Launcher.js [hunk 1]
```diff
@@ -271,9 +270,10 @@ export default class Launcher {
     launcherClient.setAppMetadata({
       sourceAccount: account._id
     })
-    const folder = await ensureKonnectorFolder(client, {
-      konnector,
-      account
+    const folder = await models.konnectorFolder.ensureKonnectorFolder(client, {
+      konnector: { ...konnector, _id: konnector._id },
```
Comment (Crash--) [inline src/libs/Launcher.js:274]: Why do you spread?

File: src/libs/Launcher.js [hunk 2]
```diff
@@ -289,6 +289,27 @@ export default class Launcher {
       trigger = triggerResponse.data
       result.createdTrigger = trigger
       log.debug(`ensureAccountAndTriggerAndJob: created trigger`, trigger)
+      // @ts-ignore
```
Comment (Crash--) [inline src/libs/Launcher.js:292]: can you add a comment to explain why you need to ignore 🙏

File: src/libs/Launcher.js (hunk 1 above)
Comment (doubleface) [reply src/libs/Launcher.js:274]: _id attribute is missing in konnector object, which causes the reference to the konnector in the destination folder to be null

I think there is something to fix in harvest when the launcher context is sent

File: src/libs/Launcher.js (hunk 2 above)
Comment (doubleface) [reply src/libs/Launcher.js]: Done, the message attribute of IOCozyTrigger is kind of a free object. Its content depends on the kind of trigger. But all the konnector trigger (client or node) which have a destination folder will have the `.message.folder_to_save` attribute.

File: src/libs/Launcher.js (hunk 1 above)
Comment (Crash--) [reply src/libs/Launcher.js]: I don't understand: 

You do: 
```
      konnector: { ...konnector, _id: konnector._id }
```

and you say : "_id attribute is missing in konnector object,". But you do: `_id: konnector._id` so _id is present in the konnector object. If this is not the case, you'll have `_id: undefined` is it really wanted?

File: src/libs/Launcher.js (hunk 2 above)
Comment (Crash--) [reply src/libs/Launcher.js]: So let's update the IOCozyTrigger type on Cozy-client 🙏

File: src/libs/Launcher.js (hunk 1 above)
Comment (doubleface) [reply src/libs/Launcher.js]: Sorry, I miss typed. The right code is `konnector: { ...konnector, _id: konnector.id }` and I fixed it now.

**Comment by doubleface** [comment]
@Crash-- Should be updated now

**Review by Crash-- — APPROVED** [review]

**Comment by doubleface** [comment]
Upgraded cozy-client with the updated triggers types and removed some @ts-ignore

**Pull request merged**